### ISSUE TITLE
`/search` page isn't customisable through-the-web

### ISSUE THREAD

**Issue by JeffersonBledsoe** [plone/volto#3465]
**Is your feature request related to a problem? Please describe.**
The current 'CMS' search page (`/search`) is a custom component. This means that the only way to customise its look or functionality is to shadow the component and implement it themselves. We already have a `Search` block which mirrors the existing functionality of this, but offers all the through-the-web customisability options of the block editor. This could either be fixed to having just a search block on the page, or a full 'block' editor to allow site admins more customisation of the search page.

**Describe the solution you'd like**
Make `/search` a block form that contains a `Search` block out-the-box.

**Comment by tiberiuichim** [comment]
@JeffersonBledsoe In one of my websites I've removed the search as a route, with 

```
  config.settings.nonContentRoutes = [
    ...config.settings.nonContentRoutes.filter((p) => p !== '/search'),
    ];
```

and I've created a /search page with a search block;

**Comment by tiberiuichim** [comment]
@JeffersonBledsoe unless we have some way of stashing the search block configuration on the backend, there's no way of having non-content routes "content aware". In principle we would need an adaptor between the Plone registry and Volto's concept of editable content.

**Comment by JeffersonBledsoe** [comment]
@tiberiuichim Just a quick thought on this. Any reason we couldn't create a normal page at `/seach` in the `plone.volto` backend package?

**Comment by tiberiuichim** [comment]
@djay Yes... but as the Search is a block and the rest of the page is basically editable, it will introduce the need for an extra settings (once this is fixed :)  https://github.com/plone/volto/blob/43ea1b68e643c53065a6fb5f613cbeb5008b0389/src/components/theme/SearchWidget/SearchWidget.jsx#L81-L83 )

**Comment by djay** [comment]
@tiberiuichim not sure what settings you refer to. 
But as long as you aren't adding text to the page then it doesn't need to be replicated. You can get away with a single root translation.
If you do add other text to the page then you would have to create copies so that extra text could be translated. But thats not really a problem is it unless you then later want to make changes to the filter settings? in which case you need to make that change in all the translations. Trying to optimise that use case by using layouts or something is probably just going to make things over complicated.

**Comment by erral** [comment]
And what about delegating this configuration to plone.volto? We can add a search page preconfigured with a search block when installing plone.volto.

**Comment by tiberiuichim** [comment]
@djay On multilingual websites the search page is exposed as /en/search, /de/search, etc, see https://github.com/plone/volto/blob/7044eca789d836786e9e789036669085cc22bee7/src/routes.js#L59-L62

The bug is that the search widget hardcodes the action as `/search`. That bug has to be fixed right now. The fact that those routes exist complicates the situation and introduces the need for a setting flag in the Volto configuration of some sort, if we want to go with /search page is persistent by default (because, if I want to edit those pages, I'd have to signal which behavior I want in the SearchWidget).

**Comment by djay** [comment]
I think the proposal here is to remove the hard coded form and route and only have a preinstalled page in the same way the news folder comes preinstalled.

**Comment by tiberiuichim** [comment]
@djay How would the following work? 

- As a multilingual website editor, 
- I want to edit the /search page,
- so that I can add additional text or multilingual content.

Now, I admit, it may be a limited scenario in real world, but it is a possibility.

So there's a balance, tipping it one way makes causes drawbacks in another place. As far as I'm concerned, I'm +1 for making the search page persistent, via plone.volto.

**Comment by djay** [comment]
@tiberiuichim i don't see it as different from another translated content. You create a translation which is essentially a copy of the page including all the search blocks. You then change what you want. The search block itself would translate itself where text is used so wouldn't need to be translated. 

The only scenario where this is a problem is if you translate your search page custom text into 100 languages and then decide after you want to reorder the filters on the side to put tags above date. In this case you would have to make this change 100 times.

Or am I missing something?

**Comment by tiberiuichim** [comment]
@djay The problem is that on multilingual websites, a translated content sits in a Language Root Folder. So, if I translate /en/my-page to Dutch, it's going to be placed in /nl/my-page. So, /search in root persistent page implies that you, as website editor, will not be able to deeply customize it, because you will not have it as a translated page.

**Comment by djay** [comment]
@tiberiuichim isn't there some kind of redirection in place when a language is not specified that takes you to the right language root? so anyone landing on /search will get redirected to /en/search ?
